internal/controller/dnsrecord_controller.go

```diff
@@ -358,14 +368,34 @@ func (r *DNSRecordReconciler) SetupWithManager(mgr ctrl.Manager, maxRequeue, val
 			}
 			return toReconcile
 		})).
+		Watches(&v1alpha1.DNSHealthCheckProbe{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+			logger := log.FromContext(ctx)
+			probe, ok := o.(*v1alpha1.DNSHealthCheckProbe)
```

Comment:
should we do a check here to see has the health of the probe changed? As I understand it on update this Watch function will be called twice (once with the old and once with the new) so if we do a Get(probe) and check status against the probe we received if that status has changed then and only then requeue?

Reply:
I guess we don't have to do that right now but if we agree lets track with a follow up issue

Reply:
Checking it here